### PR TITLE
Adding the ability for windowed CommandBarFlyouts to know how much space they have in the monitor for the purposes of opening up or down.

### PR DESCRIPTION
**Pull request by llongley**
Currently, CommandBarFlyouts that are opened unconstrained by the XAML root bounds don't have a way of knowing which monitor they're being displayed in, which means that they aren't able to get at the information regarding whether they have enough space to open their secondary commands down or whether they should open up instead.  This adds a proposed API to UIElement that can be used to retrieve that information for any given UIElement, which will give control authors the ability to act on this information.

### CODE REVIEW

File: dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp [hunk 1]
```diff
@@ -491,28 +491,43 @@ void CommandBarFlyoutCommandBar::UpdateVisualState(
...
+                    // Calling GetForCurrentView on threads without a CoreWindow throws an error. This comes up in places like LogonUI.
+                    // In this circumstance, we'll just always expand down, since we can't get bounds information.
+                }
             }
-            catch (winrt::hresult_error)
+            else
             {
-                // Calling GetForCurrentView on threads without a CoreWindow throws an error. This comes up in places like LogonUI.
-                // In this circumstance, we'll just always expand down, since we can't get bounds information.
+                winrt::IUIElement11 thisAsUIElement11 = try_as<winrt::IUIElement11>();
```
Comment (StephenLPeters) [inline dev/CommandBarFlyout/CommandBarFlyoutCommandBar.cpp:520]: These can be combined

File: dev/CommandBarFlyout/CommandBarFlyoutCommandBar.idl [hunk 2]
```diff
@@ -39,4 +39,13 @@ unsealed runtimeclass CommandBarFlyoutCommandBar : Windows.UI.Xaml.Controls.Comm
     CommandBarFlyoutCommandBarTemplateSettings FlyoutTemplateSettings { get; };
 };
 
-}
\ No newline at end of file
+// Needed so we can access this interface until it ships in the official SDK.
+[MUX_INTERNAL]
+[webhosthidden]
+[uuid(9e8f35fd-98a5-59ed-bc56-51b565acb335)]
+interface IUIElement11
```
Comment (StephenLPeters) [inline dev/CommandBarFlyout/CommandBarFlyoutCommandBar.idl:46]: Is this currently a private interface on IUIElement?

**Comment by StephenLPeters** [comment]
The CommandBarFlyout-InsufficientSpace-Open.pnglooks like it has a gap between the commandbar parts.

**Comment by llongley** [comment]
Going in a different direction - will add a PR with the API spec.

**Pull request closed without merging**